src/app/api/form/[_id]/route.ts

```diff
@@ -0,0 +1,83 @@
+import { NextRequest, NextResponse } from "next/server";
+import connectDB from "@database/db";
+import volunteerFormSchema, {
+  IVolunteerForm,
+} from "@database/volunteerFormSchema";
+
+type IParams = {
+  params: {
+    _id: string;
+  };
+};
+
+/**
+ * GET API for retrieving a volunteerForm by its id
+ * @returns None
+ */
+export async function GET(req: NextRequest, { params }: IParams) {
+  await connectDB();
+  const { _id } = params;
+
+  try {
+    const event = await volunteerFormSchema.findOne({ _id }).orFail();
+    return NextResponse.json(event, { status: 201 });
+  } catch (err: any) {
+    console.error("Error", err);
+
+    if (err.name === "DocumentNotFoundError") {
+      return NextResponse.json("No event found.", { status: 404 });
+    }
+
+    return NextResponse.json({ err: "Internal Server Error" }, { status: 500 });
+  }
+}
+
+/**
+ * PUT API for editing a volunteerForm
+ * @returns None
+ */
+export async function PUT(req: NextRequest, { params }: IParams) {
+  await connectDB();
+  const { _id } = params;
+
+  try {
+    const { eventId, questions }: IVolunteerForm = await req.json();
+    const volunteerForm = { eventId, questions };
+
+    if (volunteerForm) {
+      const updatedEvent = await volunteerFormSchema.findByIdAndUpdate(
+        { _id },
+        volunteerForm
+      );
+      return NextResponse.json(updatedEvent, { status: 201 });
+    } else {
+      console.error("Invalid request body");
+      return NextResponse.json("Invalid request body", { status: 400 });
+    }
+  } catch (error) {
+    console.error("Error creating event:", error);
+    return NextResponse.json("Internal Server Error", { status: 500 });
+  }
+}
+
+/**
+ * DELETE API for deleting a volunteerForm
+ * @returns None
+ */
+export async function DELETE(req: NextRequest, { params }: IParams) {
+  await connectDB(); // function from db.ts
+  const { _id } = params;
+
+  try {
+    const volunteerFormToDelete = await volunteerFormSchema
+      .deleteOne({ _id: _id })
+      .orFail();
+
+    return NextResponse.json(volunteerFormToDelete);
+  } catch (err) {
+    console.log(err);
+    return NextResponse.json("Unable to delete volunteerForm.", {
+      status: 404,
+    });
+  }
+}
```

Comment:
You will also need to delete the form reference from the associated Event. Do that before deleting the VolunteerForm document itself and make sure the form deletion doesn't happen unless the id reference in Event happened successfully.

Reply:
What is the form reference?

Reply:
Each Event has a single VolunteerForm associated with it. There is a form field in the Event schema reflecting this, and you'll need to delete the VolunteerForm id from that field for the associated Event document when deleting the VolunteerForm.

Reply:
When I "delete" the reference, should I just set form to "" or null? I am setting it to "" lmk if I should do otherwise

Reply:
sorry for the delayed response. I think the convention here would be just removing the "form" field from the Event document entirely. However, it's a bit weird because I can't really think of a situation where we would want to delete a Form and not the Event as well. So maybe the existence of the endpoint is the unconventional thing here lol. In the name of not wasting time on unimportant stuff I think what we have here will work for us for now.